### DISCUSSION TITLE
Calibration issue (and ghost rawZ numbers) - TFT touchscreen ST7789

### DISCUSSION THREAD

**Discussion by ArthurFR1** [Bodmer/TFT_eSPI#2958] [Q&A - Touch]
Hello!

**I got a 2.8' TFT touchscreen (driver : ST7789). The screen works perfectly** with the library and I've got no issue displaying text and examples such as TFT_Starfield.

**However, I'm facing some issues while trying to test the touchscreen.**
In fact, I am unable to get through the calibration test to get to the Keypad_240_320 example.

Tough the touch is indeed taken into account, because when I run Test_Touch_Controller, coordinates are printed in the monitor when I'm touching the screen (see the video, the X number provides zero when i'm not touching the screen, Y gives numbers between 3380 and 3320, and Z gives numbers between 2 and 26 even though I'm not touching the screen, i'm not sure it's a normal behavior).

**Please see the uploaded video (30 sec) that sums up my issue.**

https://github.com/Bodmer/TFT_eSPI/assets/149966426/580ce18a-38c0-46aa-ab00-50b32a49e2f2

I'm using :

- Arduino IDE
- Generic TFT touchscreen (with generic XPT2046 chip for touch capabilities)
- TFT_eSPI version 2.5.31
- ESP32 devboard wired to the TFT screen
- TFT's driver : ST7789

What I've done & tried :

- Raising the touch threshold to 600
- Switching SPI clock rate to 10MHz and to 1MHz
- Playing around with code and setup with no actual results
- TOUCH_CS has been defined
- Right driver (ST7789), right user setup (18), right screen dimention (240x320) have all been set up correclty.
- Code uploads with no warnings nor issues

I've been through documentation & community issues for a while now and I think I'm missing something.
Does anyone have had the same issue?
Thanking you in advance for your response.
Regards

**Comment by Bodmer** [comment]
The sketch you are running requires a memory partition to be setup for SPIFFS so I suspect that has not been done.

The Test_Touch_Controller output looks good, you can expect some "noise" on the Z (pressure) reading.

To avoid the need for a SPIFFS partition, it would be better to [use this example sketch](https://github.com/Bodmer/TFT_eSPI/blob/master/examples/Generic/Touch_calibrate/Touch_calibrate.ino) to calibrate the touch controller.

**Comment by ArthurFR1** [comment]
Hi @Bodmer , thank you for your reply.

After few days of trying to partition 3MB of SPIFFS 
![image](https://github.com/Bodmer/TFT_eSPI/assets/149966426/99eb07ee-f2a4-4e8d-abd2-386e22c9a9e0)

or event trying the Touch_calibrate.ino example sketch, it seems that either the screen is crashing when i touch it multiple times (white screen), or either it simply doesn't work (Can't calibrate nor use the touch screen for buttons/keypad/etc.).

Even though I replace the touch initialization & calibration by : `uint16_t calData[5] = { 300, 3800, 300, 3800, 6 };
  tft.setTouch(calData);`, it shows no results.
